From 2b5c1e57fccdd20edbae5e6fbce72e4955c761ec Mon Sep 17 00:00:00 2001
From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Wed, 23 Mar 2022 20:36:09 +0100
Subject: [PATCH] docs: move release notes TOC to sub page (#2422)

This allows us to organize the TOC a bit better by adding sections per
major version. We can even increase the maxdepth to 2 now, which looks
great in my opinion.

In addition, the full list of releases is not shown in the sidebar
anymore when viewing a completely different part of the documentation,
which took up more than half of the total sidebar entries.

(cherry picked from commit 7ebc88147ee96342a9de6a6774fe8f7a85925f78)
---
 docs/index.rst          |  56 +------------------
 docs/releases/index.rst | 118 ++++++++++++++++++++++++++++++++++++++++
 2 files changed, 119 insertions(+), 55 deletions(-)
 create mode 100644 docs/releases/index.rst

diff --git a/docs/index.rst b/docs/index.rst
index d8e39465..41dddaa9 100644
--- a/docs/index.rst
+++ b/docs/index.rst
@@ -78,61 +78,7 @@ Several Freifunk communities in Germany use Gluon as the foundation of their Fre
    :caption: Releases
    :maxdepth: 1
 
-   releases/v2021.1.1
-   releases/v2021.1
-   releases/v2020.2.3
-   releases/v2020.2.2
-   releases/v2020.2.1
-   releases/v2020.2
-   releases/v2020.1.4
-   releases/v2020.1.3
-   releases/v2020.1.2
-   releases/v2020.1.1
-   releases/v2020.1
-   releases/v2019.1.3
-   releases/v2019.1.2
-   releases/v2019.1.1
-   releases/v2019.1
-   releases/v2018.2.4
-   releases/v2018.2.3
-   releases/v2018.2.2
-   releases/v2018.2.1
-   releases/v2018.2
-   releases/v2018.1.4
-   releases/v2018.1.3
-   releases/v2018.1.2
-   releases/v2018.1.1
-   releases/v2018.1
-   releases/v2017.1.8
-   releases/v2017.1.7
-   releases/v2017.1.6
-   releases/v2017.1.5
-   releases/v2017.1.4
-   releases/v2017.1.3
-   releases/v2017.1.2
-   releases/v2017.1.1
-   releases/v2017.1
-   releases/v2016.2.7
-   releases/v2016.2.6
-   releases/v2016.2.5
-   releases/v2016.2.4
-   releases/v2016.2.3
-   releases/v2016.2.2
-   releases/v2016.2.1
-   releases/v2016.2
-   releases/v2016.1.6
-   releases/v2016.1.5
-   releases/v2016.1.4
-   releases/v2016.1.3
-   releases/v2016.1.2
-   releases/v2016.1.1
-   releases/v2016.1
-   releases/v2015.1.2
-   releases/v2015.1.1
-   releases/v2015.1
-   releases/v2014.4
-   releases/v2014.3.1
-   releases/v2014.3
+   releases/index
 
 License
 -------
diff --git a/docs/releases/index.rst b/docs/releases/index.rst
new file mode 100644
index 00000000..ddda8511
--- /dev/null
+++ b/docs/releases/index.rst
@@ -0,0 +1,118 @@
+Release Notes
+=============
+
+.. toctree::
+   :caption: Gluon 2021.1
+   :maxdepth: 2
+
+   v2021.1.1
+   v2021.1
+
+.. toctree::
+   :caption: Gluon 2020.2
+   :maxdepth: 2
+
+   v2020.2.3
+   v2020.2.2
+   v2020.2.1
+   v2020.2
+
+.. toctree::
+   :caption: Gluon 2020.1
+   :maxdepth: 2
+
+   v2020.1.4
+   v2020.1.3
+   v2020.1.2
+   v2020.1.1
+   v2020.1
+
+.. toctree::
+   :caption: Gluon 2019.1
+   :maxdepth: 2
+
+   v2019.1.3
+   v2019.1.2
+   v2019.1.1
+   v2019.1
+
+.. toctree::
+   :caption: Gluon 2018.2
+   :maxdepth: 2
+
+   v2018.2.4
+   v2018.2.3
+   v2018.2.2
+   v2018.2.1
+   v2018.2
+
+.. toctree::
+   :caption: Gluon 2018.1
+   :maxdepth: 2
+
+   v2018.1.4
+   v2018.1.3
+   v2018.1.2
+   v2018.1.1
+   v2018.1
+
+.. toctree::
+   :caption: Gluon 2017.1
+   :maxdepth: 2
+
+   v2017.1.8
+   v2017.1.7
+   v2017.1.6
+   v2017.1.5
+   v2017.1.4
+   v2017.1.3
+   v2017.1.2
+   v2017.1.1
+   v2017.1
+
+.. toctree::
+   :caption: Gluon 2016.2
+   :maxdepth: 2
+
+   v2016.2.7
+   v2016.2.6
+   v2016.2.5
+   v2016.2.4
+   v2016.2.3
+   v2016.2.2
+   v2016.2.1
+   v2016.2
+
+.. toctree::
+   :caption: Gluon 2016.1
+   :maxdepth: 2
+
+   v2016.1.6
+   v2016.1.5
+   v2016.1.4
+   v2016.1.3
+   v2016.1.2
+   v2016.1.1
+   v2016.1
+
+.. toctree::
+   :caption: Gluon 2015.1
+   :maxdepth: 2
+
+   v2015.1.2
+   v2015.1.1
+   v2015.1
+
+.. toctree::
+   :caption: Gluon 2014.4
+   :maxdepth: 2
+
+   v2014.4
+
+.. toctree::
+   :caption: Gluon 2014.3
+   :maxdepth: 2
+
+   v2014.3.1
+   v2014.3
+
-- 
2.17.1

